lib/emoji.rb

```diff
@@ -63,19 +63,24 @@ def edit_emoji(emoji)
 
   # Public: Find an emoji by its aliased name. Return nil if missing.
   def find_by_alias(name)
-    names_index[name]
+    @mutex.synchronize do
```

Comment:
Could you instead add the mutex guard around the code in `all` method? That way we can have the `synchronize` call only in one place.